prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/FHIRBundleHelpersTests.kt

```diff
@@ -688,4 +693,136 @@ class FHIRBundleHelpersTests {
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_with_birth_time.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.handleBirthTime(hl7messages[0])
+
+        val patient = FhirPathUtils.evaluate(
+            CustomContext(bundle, bundle),
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(Patient)"
+        )[0] as Patient
+
+        assertThat(patient).isNotNull()
+        val birthDateTimeValue = patient.birthDateElement.extension[0].value.primitiveValue()
+        assertThat(
+            birthDateTimeValue
+        ).isEqualTo("2023-05-04T13:10:23-05:00")
+    }
+
+    @Test
+    fun `Test enhance bundle metadata 2-5-1`() {
+        // set up
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/fhir_without_patient.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        assertThat(messages).isNotEmpty()
+        val bundle = messages[0]
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_with_birth_time.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.enhanceBundleMetadata(hl7messages[0])
+
+        assertThat(bundle.timestamp.toString()).isEqualTo("Wed Feb 10 17:07:37 PST 2021")
+        assertThat(bundle.identifier.value).isEqualTo("371784")
+        assertThat(bundle.identifier.system).isEqualTo("https://reportstream.cdc.gov/prime-router")
+    }
+
+    @Test
+    fun `Test enhance bundle metadata 2-7`() {
+        // set up
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/fhir_without_patient.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        assertThat(messages).isNotEmpty()
+        val bundle = messages[0]
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_2.7.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.enhanceBundleMetadata(hl7messages[0])
+
+        assertThat(bundle.timestamp.toString()).isEqualTo("Wed Feb 10 17:07:37 PST 2021")
+        assertThat(bundle.identifier.value).isEqualTo("371785")
+        assertThat(bundle.identifier.system).isEqualTo("https://reportstream.cdc.gov/prime-router")
+    }
+
+    @Test
+    fun `Test enhance bundle metadata 2-6`() {
```

Comment:
2.6 is just a sample use case, the purpose of this test isn't to test 2.6 specifically, right? It's to test an "unsupported message ID", or whatever language you want to use, so I would rename this "Test enhance bundle unsupported message ID" or something.

Reply:
Renamed this test 👍